openedx/core/djangoapps/ratelimit/middleware.py

```diff
@@ -0,0 +1,8 @@
+from django.conf import settings
+from ipware.ip import get_ip
+
+
+class ReverseProxy(object):
+    def process_request(self, request):
+        if settings.RATELIMIT_ENABLE:
+            request.META['REMOTE_ADDR'] = get_ip(request)
```

Comment:
Would you be up for using callable values with a custom key method (e.g. `real_ip`), as documented in:

    http://django-ratelimit.readthedocs.io/en/v1.1.0/keys.html#callable-values

I'd rather not have an extra middleware run, particularly as `ipware.ip` examines the `REMOTE_ADDR` as part of it's work, and I'm worried about confusion and odd side-effects down the road.

Reply:
If you want to keep the approach you have now, that's okay too. But please put some comments explaining why this file exists and what problem it's solving.